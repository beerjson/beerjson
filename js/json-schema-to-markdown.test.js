const parse = require('../js/json-schema-to-markdown.js')
const typeSchema = require('../json/fermentable.json')
const objSchema = require('../json/beer.json')

test('fermentable docs should match snapshot', () => {
  expect(parse(typeSchema)).toMatchSnapshot()
})

test('formatTypeDefinition', () => {
  expect(
    parse.formatTypeDefinition([
      'FermentableBase',
      typeSchema.definitions.FermentableBase
    ])
  ).toMatchSnapshot()
})
