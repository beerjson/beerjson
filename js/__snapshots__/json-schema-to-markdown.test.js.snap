// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fermentable docs should match snapshot 1`] = `
"The schema defines the following types:

## FermentableBase 

FermentableBase provides unique properties to identify individual records of fermentable ingredients

\`FermentableBase\` type: \`object\`

### Properties

|   |Type|Description|Required|
|---|----|-----------|--------|
| **name** | string|  | :white_check_mark: |
| **type** |  'dry extract'  , 'extract'  , 'grain'  , 'sugar'  , 'fruit'  , 'juice'  , 'honey'  , 'other' |  | :white_check_mark: |
| **origin** | string|  |  |
| **supplier** | string|  |  |
| **grain_group** |  'base'  , 'caramel'  , 'flaked'  , 'roasted'  , 'specialty'  , 'smoked'  , 'adjunct' |  |  |

## FermentableType 

FermentableType collects the attributes of a fermentable ingredient to store as record information

\`FermentableType\` type: \`object\`

Parent: [FermentableBase](#fermentablebase)

### Properties

|   |Type|Description|Required|
|---|----|-----------|--------|
| **yield** | object|  | :white_check_mark: |
| **bitterness** | object|  |  |
| **notes** | string|  |  |
| **moisture** | [PercentType](measureable_units.json.md#percenttype)|  |  |
| **alpha_amylase** | number|  |  |
| **diastatic_power** | [DiastaticPowerType](measureable_units.json.md#diastaticpowertype)|  |  |
| **protein** | [PercentType](measureable_units.json.md#percenttype)|  |  |
| **soluble_nitrogen_ratio** | number|  |  |
| **max_in_batch** | [PercentType](measureable_units.json.md#percenttype)|  |  |
| **recommend_mash** | boolean|  |  |
| **color** | [ColorType](measureable_units.json.md#colortype)|  | :white_check_mark: |
| **inventory** | object|  |  |

## FermentableAdditionType 

FermentableAdditionType collects the attributes of each fermentable ingredient for use in a recipe fermentable bill

\`FermentableAdditionType\` type: \`object\`

Parent: [FermentableBase](#fermentablebase)

### Properties

|   |Type|Description|Required|
|---|----|-----------|--------|
| **timing** | object|  |  |
| **amount** | undefined|  |  |

"
`;

exports[`formatTypeDefinition 1`] = `
"## FermentableBase 

FermentableBase provides unique properties to identify individual records of fermentable ingredients

\`FermentableBase\` type: \`object\`

### Properties

|   |Type|Description|Required|
|---|----|-----------|--------|
| **name** | string|  | :white_check_mark: |
| **type** |  'dry extract'  , 'extract'  , 'grain'  , 'sugar'  , 'fruit'  , 'juice'  , 'honey'  , 'other' |  | :white_check_mark: |
| **origin** | string|  |  |
| **supplier** | string|  |  |
| **grain_group** |  'base'  , 'caramel'  , 'flaked'  , 'roasted'  , 'specialty'  , 'smoked'  , 'adjunct' |  |  |

"
`;
